apps/web/src/components/Shared/Login/New/index.tsx

```diff
@@ -50,45 +58,78 @@ const NewProfile: FC<NewProfileProps> = ({ isModal = false }) => {
         to: address,
         handle: debouncedHandle,
         imageURI: avatar,
-        followModule: '0x0000000000000000000000000000000000000000',
+        followModule: ZERO_ADDRESS,
         followModuleInitData: '0x',
         followNFTURI: ''
       }
     ],
     enabled: Boolean(debouncedHandle)
   });
-  const { data, write } = useContractWrite(config);
+  const { data: txData, write } = useContractWrite(config);
 
   const { isLoading, error } = useWaitForTransaction({
-    hash: data?.hash
+    hash: txData?.hash
   });
 
-  /*const handleUpload = async (evt: ChangeEvent<HTMLInputElement>) => {
-                                                                          evt.preventDefault();
-                                                                          setUploading(true);
-                                                                          try {
-                                                                            const attachment = await uploadToIPFS(evt.target.files);
-                                                                            if (attachment[0]?.item) {
-                                                                              setAvatar(attachment[0].item);
-                                                                            }
-                                                                          } finally {
-                                                                            setUploading(false);
-                                                                          }
-                                                                        };*/
+  const handleUpload = async (evt: ChangeEvent<HTMLInputElement>) => {
+    evt.preventDefault();
+    setUploading(true);
+    try {
+      const attachment = await uploadToIPFS(evt.target.files);
+      if (attachment[0]?.item) {
+        setAvatar(attachment[0].item);
+      }
+    } finally {
+      setUploading(false);
+    }
+  };
 
   const handleCreateProfile = async () => {
     write?.();
   };
 
-  return (
-    <Form form={form} className="space-y-4" onSubmit={handleCreateProfile}>
-      {error && (
+  useEffect(() => {
+    setIsCreationLoading(
+      IS_RELAYER_AVAILABLE ? data?.createProfile.__typename === 'RelayerResult' : !!txData?.hash
```

Comment:
Why don't we use `isLoading` from `useWaitForTransaction` instead of `!!txData?.hash`?

Reply:
Technically, this should give the same result.
Philosophically, I wanted to be sure we had the transaction hash so that the `Pending` component worked well.
What do you think?

Reply:
Ah ok got it! Agreed 👍